toffy/streak_detection_test.py

```diff
@@ -0,0 +1,445 @@
+from typing import Tuple
+import pytest
+from pathlib import Path
+from toffy import streak_detection as sd
+import numpy as np
+import pandas as pd
+from skimage import io
+from collections import namedtuple
+import xarray as xr
+from ark.utils import test_utils
+from functools import partial
+
+
+@pytest.fixture(scope="function")
+def streak_dataset():
+    def streak_data_generator(corrected_dir, fov: str, chan: str, shape: Tuple[int, int]):
+        _test_data = sd.StreakData(
+            shape=shape,
+            fov=fov,
+            corrected_dir=corrected_dir,
+            streak_channel=chan,
+            streak_mask=np.zeros(shape=shape),
+            streak_df=pd.util.testing.makeDataFrame(),
+            filtered_streak_mask=np.zeros(shape=shape),
+            filtered_streak_df=pd.util.testing.makeDataFrame(),
+            boxed_streaks=np.zeros(shape=shape),
+            corrected_streak_mask=np.zeros(shape=shape),
+        )
+        return _test_data
+
+    return streak_data_generator
+
+
+@pytest.fixture(scope="function")
+def df_dataset():
+    def test_df_generator(shape: Tuple[int, int], df_row_num: int):
+        col_size = shape[0]
+        row_size = shape[1]
+
+        # Initialize a new generator - set seed for reproducibility
+        rng = np.random.default_rng(12345)
+
+        # Create test_df
+        min_row = rng.integers(low=0, high=row_size - 1, size=(df_row_num))
+        max_row = min_row + 1
+        min_col = rng.integers(low=0, high=col_size - 1, size=(df_row_num))
+        max_col = rng.integers(low=min_col, high=row_size - 1, size=(df_row_num))
+        test_df = pd.DataFrame(
+            np.stack([min_row, max_row, min_col, max_col], axis=-1),
+            columns=["min_row", "max_row", "min_col", "max_col"],
+        )
+        return test_df
+
+    return test_df_generator
+
+
+def test_get_save_dir(tmp_path):
+    data_dir: Path = Path(tmp_path)
+    name: str = "streak_data_test"
+    ext_csv: str = "csv"
+    ext_tiff: str = "tiff"
+
+    csv_save_dir: Path = sd._get_save_dir(data_dir=data_dir, name=name, ext=ext_csv)
+    assert csv_save_dir == Path(tmp_path / "streak_data_test.csv")
+
+    tiff_save_dir: Path = sd._get_save_dir(data_dir=data_dir, name=name, ext=ext_tiff)
+    assert tiff_save_dir == Path(tmp_path / "streak_data_test.tiff")
+
+
+def test_save(tmp_path, streak_dataset):
+    # Create minimum data needed for testing the save functionality with dataframes and numpy array
+    # streak_data needs: corrected_dir, streak_channel, and data (streak_mask, streak_df,
+    # filtered_streak_mask, filtered_streak_df, boxed_streaks, corrected_streak_mask)
+
+    streak_data_test = streak_dataset(
```

Comment:
For testing it's good to use much smaller images (like 20x20) wherever possible to speed things